---
title: "'5 accessions' RNAseq results"
output:
  ioslides_presentation:
    standalone: true
    self_contained: true
    smaller: true
    transition: "faster"
    widescreen: true
    css: templates/source2.css
    fig_width: 9.16
---

```{r setup, include=FALSE, results="hide"}
library(ggplot2)
library(data.table)
library(grid)
library(gridExtra)
theme_slide <- theme_grey(base_size = 16, base_family = "Lato") +
  theme(plot.background = element_rect(fill = "transparent", colour = NA),
        legend.background = element_rect(fill = "transparent", colour = NA))
knitr::opts_chunk$set(
  dev.args=list(bg = "transparent", family = "Lato"),
  echo = FALSE,
  message = FALSE,
  warning = FALSE
  )
source("src/figures.R")
```

## Phenotypic diversity

```{r diversity}
img <- png::readPNG("data/diversity.png")
grid::grid.raster(img)
```

## Panicle phenotypes {.centered .vcenter}

```{r phenotypes}
ggplot(plot.data.phenotypes, aes(x = Accession, y = value, colour = Species)) +
  theme_slide +
  theme(
    axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1),
    strip.background = element_blank(),
    legend.text = element_text(face = "italic")
    ) +
  xlab(NULL) + ylab(NULL) +
  scale_color_brewer(palette = "Set1", guide = guide_legend(title=NULL)) +
  facet_wrap(~variable, switch = 'y', scales = "free_y") +
  geom_point(position = position_jitter(width = 0.5), size = 3, alpha = 0.75)
```

## Sequencing stages

```{r stages}
grid::grid.newpage()
grid::pushViewport(grid::viewport(layout = grid::grid.layout(1, length(pngs))))
for (i in 1:length(pngs)){
grid::grid.raster(pngs[[i]], width = 0.95,
                  vp = grid::viewport(layout.pos.row = 1, layout.pos.col = i))
}
```

## Sequencing results {.centered .vcenter}

```{r stats}
ggplot(plot.data.stats, aes(x = `Input reads (M)`, y = value,
                            colour = Accession)) +
  theme_slide +
  theme(strip.background = element_blank(),
        legend.position = "top") +
  scale_color_brewer(palette = "Set1", guide = guide_legend(title=NULL)) +
  facet_wrap(~variable, scales = "free", switch = "y") +
  ylab(NULL) +
  geom_point() +
  geom_smooth(method = "lm", se = FALSE)
```

## Differential expression analysis with GLMs

* Generalized linear model (GLM):
    * $Y_i = b_0 + b_1 X_{1i} + \dots + b_n X_{ni} + \varepsilon_i$
* Describe gene expression (read counts) with a GLM
    * $\text{RC}_i \propto X_{1i} + \dots + X_{ni}$
    * In `R` (`DESeq2`, `EdgeR`, `limma`, ...): `design = ~ X1 + X2`
* Simple case: $\text{GE}_i \propto \text{accession}_i + \text{stage}_i$
    * `design = ~ accession + stage`    
```{r specColData}
head(colData(dds.species))
```

##

```{r waldspecies, results = "asis"}
pander::pandoc.table(
    species.contrast.results[, .(
      `|L~2~FC| > 1 ( *p*~adj~ < 0.05 )` = sum(padj < 0.05, na.rm = TRUE),
      `Up` = sum(padj < 0.05 & log2FoldChange > 0, na.rm = TRUE),
      `Down` = sum(padj < 0.05 & log2FoldChange < 0, na.rm = TRUE)),
    by = contrast], "design = ~ accession + stage", emphasize.strong.cols = c(1))
```

## design = ~ accession + stage {.centered .vcenter}

```{r speciesill}
grid::grid.draw(arrangeGrob(grobs = plots, ncol = 5))
```

## GLM with interaction

$\text{RC}_i \propto \text{accession}_i + \text{stage}_i + \text{stage}_i:\text{accession}_i$

* The effect of `stage` on `RC` is **different** at different levels of the `accession` factor.
* Lots of genes, more on this another time.

##

### design = ~ accession + stage + stage:domestication

```{r domColData}
colData(dds.domestication)
```

* Are there genes where the difference in expression between PBM and SM depends on domestication status?
* 13 genes with *p*~adj~ < 0.05...

## stage:domestication {.centered .vcenter}

```{r domPlot}
ggplot(sig.domestication.tpm, aes(x = stage, y = tpm, colour = Accession,
                                  group = Accession)) +
  theme_grey(base_size = 10, base_family = "Lato") +
  theme(plot.background = element_rect(fill = "transparent", colour = NA),
        legend.background = element_rect(fill = "transparent", colour = NA),
        legend.position = "top") +
  xlab(NULL) + ylab("Expression (TPM)") + 
  facet_wrap(~gene.name, scales = "free_y", nrow = 3) +
  scale_colour_brewer(palette = "Set1", guide = guide_legend(title=NULL)) +
  geom_smooth(method = "lm", se = FALSE, size = 0.5) +
  geom_point(position = position_jitter(width = 0.2), alpha = 0.75)
```

## Domestication by continent

* Same as the `stage:domestication` interaction but within continents
    * *i.e.* *O. sativa* *vs*. *O. rufipogon* and *O. glaberrima* *vs*. *O. barthii*
* Use a combination of continent and domestication as a grouping factor
```{r dpcColData}
colData(dds.domestication.by.continent)
```
* `~ stage + group + group:stage`

## Asia: `r dim(results.table.dbc[domestication == "asia" & padj < 0.05])[1]` genes with *p*~adj~ < 0.05 {.centered .vcenter}

```{r domAsia}
ggplot(dbc.asia[gene.name %in% levels(gene.name)[c(1:9)]],
       aes(x = stage, y = tpm, colour = Accession, group = Accession)) +
  theme_grey(base_size = 12, base_family = "Lato") +
  theme(plot.background = element_rect(fill = "transparent", colour = NA),
        legend.background = element_rect(fill = "transparent", colour = NA),
        legend.position = "top") +
  xlab(NULL) + ylab("Expression (TPM)") + 
  facet_wrap(~gene.name, scales = "free_y", nrow = 3) +
  scale_colour_brewer(palette = "Set1", guide = guide_legend(title=NULL)) +
  geom_smooth(method = "lm", se = FALSE, size = 0.5) +
  geom_point(position = position_jitter(width = 0.2), alpha = 0.75)
```

## Africa: `r dim(results.table.dbc[domestication == "africa" & padj < 0.05])[1]` genes with *p*~adj~ < 0.05 {.centered .vcenter}

```{r domAfrica}
ggplot(dbc.africa[gene.name %in% levels(gene.name)[c(1:9)]],
       aes(x = stage, y = tpm, colour = Accession, group = Accession)) +
  theme_grey(base_size = 12, base_family = "Lato") +
  theme(plot.background = element_rect(fill = "transparent", colour = NA),
        legend.background = element_rect(fill = "transparent", colour = NA),
        legend.position = "top") +
  xlab(NULL) + ylab("Expression (TPM)") + 
  facet_wrap(~gene.name, scales = "free_y", nrow = 3) +
  scale_colour_manual(values = RColorBrewer::brewer.pal(5, "Set1")[c(4, 5)],
                      guide = guide_legend(title=NULL)) +
  geom_smooth(method = "lm", se = FALSE, size = 0.5) +
  geom_point(position = position_jitter(width = 0.2), alpha = 0.75)
```

## ERF genes (AP2 transcription factors) {.centered .vcenter}

```{r erf}
gp <- ggplot(erfs.tpm, aes(x = stage, y = tpm, colour = Accession,
                     group = Accession)) +
  theme_grey(base_size = 12, base_family = "Lato") +
  theme(plot.background = element_rect(fill = "transparent", colour = NA),
        legend.background = element_rect(fill = "transparent", colour = NA)) +
  xlab(NULL) + ylab("Expression (TPM)") + 
  facet_wrap(~gene.name, scales = "free_y") +
  scale_colour_brewer(palette = "Set1", guide = guide_legend(title=NULL)) +
  geom_smooth(method = "lm", se = FALSE, size = 0.5, alpha = 0.5) +
  geom_point(position = position_jitter(width = 0.2))

img <- png::readPNG("data/mfs1.png")

grid::grid.newpage()
grid::pushViewport(grid::viewport(layout = grid::grid.layout(4, 1)))
print(gp, vp = grid::viewport(layout.pos.row = 1:3, layout.pos.col = 1))
grid::grid.raster(img, vp = grid::viewport(layout.pos.row = 4, layout.pos.col = 1))

```

## Now what?

* Confirm expression (PCR, *in situ*, etc.)
* Look for SNPs in rice 3000 genomes / African re-sequencing data
* Other ideas?
